lib/iris/tests/test_concatenate.py

```diff
@@ -748,6 +730,49 @@ def test_concat_2x2d_aux_xy_bounds(self):
         self.assertEqual(len(result), 1)
         self.assertEqual(result[0].shape, (2, 4))
 
+    def test_fill_value_invariant_to_order__same_non_None(self):
```

Comment:
This test naming was a bit confusing to me. I only understood what `__same_non_None` meant when I started writing a comment asking you what it is. I am going to try to think of a different name